app/src/main/java/dev/arkbuilders/arkmemo/media/ArkAudioRecorderImpl.kt

```diff
@@ -47,13 +49,24 @@ class ArkAudioRecorderImpl @Inject constructor(
 
     override fun stop() {
         recorder?.let {
-            it.stop()
+            try {
+                it.stop()
+            } catch (e: RuntimeException) {
+                Log.e(TAG, "stop exception: " + e.message)
+            }
+
             it.release()
         }
         recorder = null
     }
 
-    override fun maxAmplitude(): Int = recorder?.maxAmplitude!!
+    override fun maxAmplitude(): Int {
+        try {
+            return recorder?.maxAmplitude ?: 0
+        } catch (e: Exception) {
+            return 0
+        }
```

Comment:
```suggestion
            return recorder?.maxAmplitude ?: 0
```

Reply:
@hieuwu `maxAmplitude` could throw an exception if it is called before the audio source has been set.
Any reason why you think we can get rid of the `try catch` block?
 